réglage de la diffusion du bouton

MohamedRayenJmili opened the pull request:
Pull Request Description:

This pull request addresses a few issues related to the display and functionality of the "Retrieve Funds" button. The changes made are as follows:

    Adjusted the button display:
        Modified the CSS styles to ensure that the button is always visible, regardless of scrolling behavior.
        Removed any conditions that were causing the button to be hidden or displayed inconsistently.

    Improved instant display:
        Resolved an issue where the button was not showing immediately upon page load or component initialization.
        Adjusted the code to ensure that the button is displayed instantly without any delay.

    Fixed premature button enablement:
        Addressed a bug where the "Retrieve Funds" button was enabled even before the 15-day time period elapsed.
        Updated the logic to correctly determine when the button should be enabled based on the 15-day timeout.

These changes improve the user experience by ensuring that the button is always displayed, shown instantly, and enabled only when it should be according to the specified time constraints.

Please review and merge this pull request to incorporate the necessary fixes and enhancements.

Commit:
réglage de la diffusion du bouton

hichri-louay commented on src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.html:
```diff
@@ -438,13 +438,14 @@ <h1 class="campaign-brand">
           id="btn-edit-refunds-pool"
           (click)="getRefunds(campaign.id)"
-          *ngIf="campaign.isOwnedByUser && campaign.type === 'finished'  && !loadingData && campaign.budget !== '0'"
+          *ngIf="campaign.type === 'finished'  && !loadingData && campaign.budget !== '0'"
```
I noticed in your recent changes that the campaign.isOwnedByUser condition has been removed. As per my understanding, this function is meant to be available only for the owner of the campaign.

Could you please help me understand why this condition was removed? Is there another check in place to ensure that the function remains accessible only to the campaign owner?

hichri-louay commented on src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts:
```diff
@@ -271,27 +271,7 @@ export class CampaignDetailComponent implements OnInit {
       this.getKits();
     }
     
-    setTimeout(() => {
```
I noticed in your recent changes that the setTimeout function has been removed. My understanding is that we were using this timeout as a means to delay the execution until the data from the API was available.

Is there an alternative method in place to handle the asynchronous data fetching from the API?

hichri-louay commented on src/app/campaigns/campaign-details/components/campaign-detail/campaign-detail.component.ts:
```diff
@@ -271,27 +271,7 @@ export class CampaignDetailComponent implements OnInit {
-      // CALCULATE MINUTES : 
-      this.dateRefundMinutes = Math.floor(((((this.dateRefund.getTime() - Date.now()) / (1000 * 60 * 60 * 24  ) - this.dateRefundDays ) * 24) - this.dateRefundHours) * 60)
-      this.loadingData = false;
-    }, 3000)
```
I noticed in your recent changes that the setTimeout function has been removed. My understanding is that we were using this timeout as a means to delay the execution until the data from the API was available.

Is there an alternative method in place to handle the asynchronous data fetching from the API?

HamdiBenK replied:
yes i am the one who removed it and i have putted the logic in the function that fetch the data, the setTimeout function was causing problems

Commit:
correction de quelques mauvaises pratiques

Commit:
Correction de l'affichage du bouton

Commit:
Refactor button display to match Figma design

MohamedRayenJmili commented:
Pull Request Description:

This pull request addresses a few issues related to the display and functionality of the "Retrieve Funds" button. The changes made are as follows:

Adjusted the button display:
    Modified the CSS styles to ensure that the button is always visible, regardless of scrolling behavior.
    Removed any conditions that were causing the button to be hidden or displayed inconsistently.

Improved instant display:
    Resolved an issue where the button was not showing immediately upon page load or component initialization.
    Adjusted the code to ensure that the button is displayed instantly without any delay.

These changes improve the user experience by ensuring that the button is always displayed, shown instantly, and enabled only when it should be according to the specified time constraints.

Please review and merge this pull request to incorporate the necessary fixes and enhancements.

HamdiBenK committed:
Merge branch 'develop' into 329-button-retreive-budgetfinished-adpool
